tests/git.zunit

```diff
@@ -0,0 +1,319 @@
+        assert ${actual[2]} same_as "$(tput setaf 3)master         $(tput sgr0) 1st commit"
+        assert ${actual[3]} same_as "$(tput setaf 3)v1             $(tput sgr0) 1st commit"
+        assert ${actual[4]} same_as "$(tput setaf 3)v2             $(tput sgr0)  2nd commit"
+        assert ${actual[5]} same_as "$(tput setaf 3)$hash_1        $(tput sgr0) 1st commit"
+    }
+
+    LBUFFER=(git checkout '**')
+    prefix=
+    _fzf_complete_git 'git checkout '
+
+    LBUFFER=(git log '**')
+    prefix=
+    _fzf_complete_git 'git log '
+
+    LBUFFER=(git rebase '**')
+    prefix=
+    _fzf_complete_git 'git rebase '
+
+    LBUFFER=(git reset '**')
+    prefix=
+    _fzf_complete_git 'git reset '
+}
+
+@test 'Testing git branch/cherry-pick/merge' {
+    _fzf_complete() {
+        assert $# equals 2
+        assert $1 same_as '--ansi --tiebreak=index --multi'
+        assert $2 matches '^git (branch|cherry-pick|merge) '
+
+        hash_1=$(git rev-parse --short v1)
+
+        input=$(cat)
+        actual=(${(f)input})
+
+        assert ${#actual} equals 5
+        assert ${actual[1]} same_as "$(tput setaf 3)another-branch $(tput sgr0)  2nd commit"
+        assert ${actual[2]} same_as "$(tput setaf 3)master         $(tput sgr0) 1st commit"
+        assert ${actual[3]} same_as "$(tput setaf 3)v1             $(tput sgr0) 1st commit"
+        assert ${actual[4]} same_as "$(tput setaf 3)v2             $(tput sgr0)  2nd commit"
+        assert ${actual[5]} same_as "$(tput setaf 3)$hash_1        $(tput sgr0) 1st commit"
+    }
+
+    LBUFFER=(git branch -D '**')
+    prefix=
+    _fzf_complete_git 'git branch -D '
+
+    LBUFFER=(git cherry-pick '**')
+    prefix=
+    _fzf_complete_git 'git cherry-pick '
+
+    LBUFFER=(git merge '**')
+    prefix=
+    _fzf_complete_git 'git merge '
+}
+
+@test 'Testing git commit --fixup=/--reedit-message=/--reuse-message=/--squash=' {
+    _fzf_complete() {
+        assert $# equals 2
+        assert $1 same_as '--ansi --tiebreak=index '
+        assert $2 same_as 'git commit '
+
+        hash_1=$(git rev-parse --short v1)
+
+        input=$(cat)
+        actual=(${(f)input})
+
+        assert ${#actual} equals 5
+        assert ${actual[1]} same_as "$(tput setaf 3)${prefix}another-branch $(tput sgr0)  2nd commit"
+        assert ${actual[2]} same_as "$(tput setaf 3)${prefix}master         $(tput sgr0) 1st commit"
+        assert ${actual[3]} same_as "$(tput setaf 3)${prefix}v1             $(tput sgr0) 1st commit"
+        assert ${actual[4]} same_as "$(tput setaf 3)${prefix}v2             $(tput sgr0)  2nd commit"
+        assert ${actual[5]} same_as "$(tput setaf 3)${prefix}$hash_1        $(tput sgr0) 1st commit"
+    }
+
+    LBUFFER=(git commit '--fixup=**')
+    prefix=--fixup=
+    _fzf_complete_git 'git commit '
+
+    LBUFFER=(git commit '--reedit-message=**')
+    prefix=--reedit-message=
+    _fzf_complete_git 'git commit '
+
+    LBUFFER=(git merge '--reuse-message=**')
```

Comment:
It is `commit`, not `merge`.
There are three other places.

Reply:
😭